src/java/org/apache/cassandra/index/sai/disk/v2/sortedterms/SortedTermsReader.java

```diff
@@ -115,15 +115,10 @@ public long ceiling(@Nonnull ByteComparable term)
     {
         Preconditions.checkNotNull(term, "term null");
 
-        try (TrieRangeIterator reader = new TrieRangeIterator(termsTrie.instantiateRebufferer(),
-                                                              meta.trieFP,
-                                                              term,
-                                                              null,
-                                                              true,
-                                                              true))
+        try (TrieTermsDictionaryReader reader = new TrieTermsDictionaryReader(termsTrie.instantiateRebufferer(), meta.trieFP))
```

Comment:
We should move `ceiling`, `floor` & `getExactPointId` down to the `Cursor`. These methods are only called from the `PrimaryKeyMap` which is `NotThreadSafe`. We could then create a `TrieTermsDictionaryReader` for the `Cursor` and not have to create one for each call to these methods. 

This change was made to the OS code for the same reason.

Reply:
Sure, that makes sense to me. Do you have easy access to that work upstream? I can try to match it if that makes sense.